client/src/components/relatedProducts/ProductCard.jsx

```diff
@@ -22,7 +22,31 @@ const ProductCard = (props) => {
         category: props.info[i].category,
         description: props.info[i].description,
         defaultPrice: props.info[i].default_price,
-        features: props.info[i].features
+        features: props.info[i].features,
+        salePrice: items.results[0].sale_price
+      })
+    }
+  }
+
+  let thisOne = relevantProps.reduce((map, value) => {
+    map[value.id] = value;
+    return map;
+  }, {});
+  let allAccumulatedValues = [];
+
+  for (let j = 0; j < props.relatedRatings.length; j++) {
+    let items = thisOne[props.relatedRatings[j].id];
+    if (items) {
+      allAccumulatedValues.push({
+        id: items.id,
+        results: items.results,
+        name: items.name,
+        category: items.category,
+        description: items.description,
+        defaultPrice: items.defaultPrice,
+        features: items.features,
+        salePrice: items.results[0].sale_price,
+        rating: props.relatedRatings[j].avgRating.ratingPercentage
```

Comment:
I know I saw this like every time, but I'm still so impressed with your reduce function!